clusterDeployer.py

```diff
@@ -309,7 +310,17 @@ def ensure_infraenv_created(self, *, is_workers: bool) -> str:
 
         return infra_env
 
+    def nodes_set_password(self, *, is_workers: bool, hosts: Iterable[ClusterHost]) -> None:
+        nodes = []
+        for h in hosts:
+            nodes.extend(h.k8s_worker_nodes if is_workers else h.k8s_master_nodes)
```

Comment:
What is this doing in the function to `nodes_set_password`. If there are surprises in the code, then the code is not easy to read.

Reply:
To elaborate, the fact that you are passing in a boolean to filter nodes in hosts. That doesn't belong in a function that is called `nodes_set_password`.

Reply:
I drop this patch.

Reply:
I'm not saying we shouldn't have a set_password on a node. Rather, I'm saying `nodes_set_password` shouldn't do all of the things the your current patch does.